### PR TITLE
Validation Gas Fees

### PR DESCRIPTION
**Pull request by l-monninger**
# Summary
- **RFCs**: $\emptyset$.
- **Categories**: `protocol-units`
- **Associated PRs**: https://github.com/movementlabsxyz/aptos-core/pull/81

Integrates `validation_only` gas fee into `maptos_opt_executor` and adds tests. 

# Testing

1. Unit and e2e tests covering gas charges. 

# Outstanding issues
1. Tests are still WIP.

### CODE REVIEW

File: protocol-units/execution/opt-executor/src/executor/execution.rs [hunk 1]
```diff
@@ -260,6 +265,24 @@ mod tests {
 		SignedTransaction::new(raw_transaction, public_key, Ed25519Signature::dummy_signature())
 	}
 
+	fn create_signed_transaction_for_account(
```
Comment (0xPrimata) [inline protocol-units/execution/opt-executor/src/executor/execution.rs:268]: This seems like the only significant chunk of code that should be performing validation, but I'm not exactly seeing gas fees for validation 🤔

File: protocol-units/execution/opt-executor/src/executor/execution.rs [hunk 2]
```diff
@@ -476,4 +499,82 @@ mod tests {
...
+		let (tx_sender, _tx_receiver) = mpsc::channel(16);
+		let (executor, _config, _tempdir) = Executor::try_test_default(private_key)?;
+		let (context, _transaction_pipe) = executor.background(tx_sender)?;
+		let service = Service::new(&context);
+		executor.rollover_genesis_now().await?;
+
+		// Initialize a root account using a predefined keypair and the test root address.
+		let root_account = LocalAccount::new(
+			aptos_test_root_address(),
+			AccountKey::from_private_key(context.config().chain.maptos_private_key.clone()),
```
Comment (andyjsbell) [inline protocol-units/execution/opt-executor/src/executor/execution.rs:516]: We may want to fix this smelly one at some point in a new PR

**Review by andyjsbell — APPROVED** [review]
a small nit but lgtm

**Comment by l-monninger** [comment]
Marking this as draft for reasons discussed internally.

**Pull request closed without merging**